### ISSUE TITLE
Prior participation eligibility: ignore empty responses, add tests

### ISSUE THREAD

**Issue by becky-gilbert** [lookit/lookit-api#1302]
# TL;DR

When checking the prior study participation criteria, when the study in question is internal, we should ignore empty responses (i.e. sequence = []). We should also add more tests around the eligibility checks/warnings.

# Narrative

We recently added the ability for researchers to add prior study participation criteria to their studies, and we included this in the set of checks that are used to produce the 'ineligible' warning that appears on the participation page when an ineligible child is selected. However, the 'ineligible' warning text appears when the child has an 'empty' study response (e.g. they click 'participate now', which generates an empty response object, but then they immediately end the study). This is confusing for families.

We also need to make sure that we're testing all edge cases (e.g. empty responses) and combinations of study eligibility criteria (age range, criteria expression, prior participation). These tests can be done specifically on the study eligibility logic, and/or incorporated into the existing tests for things that are affected by eligibility (e.g. recruitment emails, study list view with child filter, eligibility warning text).

# Acceptance Criteria

- [x] Study participation checks should ignore responses from internal studies that are totally empty
- [x] Tests: 
  - Empty responses for internal studies should not count as having 'participated'
  - Partial/incomplete and completed responses for internal studies do count as 'participated'
  - Responses to external studies are always counted as 'participated'
  - ~~Check that these things are true anywhere the study criteria checks are used: recruitment emails, study list view, study-details view (warnings)~~ (I decided not to add these tests because they use the same few functions that are now well tested)
- [x] Alternatives Considered: we considered using the response's `completed` field in the prior participation checks, but decided against it because it's not a reliable flag and there's no corresponding flag for external studies. Researchers can choose to implement their own checks using the protocol generator function (past sessions data).
- [x] Documentation: in the section about study create/edit fields, explain how prior participation is checked, including edge cases, and point researchers to alternatives (e.g. get past sessions from protocol generator): https://github.com/lookit/ember-lookit-frameplayer/pull/350, https://github.com/lookit/lookit-docs/pull/365

**Comment by becky-gilbert** [comment]
Closed via #1307
